lib/clog/clog.coffee

```diff
@@ -0,0 +1,114 @@
+_ = require 'underscore'
+kv  = require "../kayvee"
+
+LOG_LEVELS = {
+  Debug:    "debug",
+  Info:     "info",
+  Warning:  "warning",
+  Error:    "error",
+  Critical: "critical",
+}
+
+LOG_LEVEL_ENUM = {
+  debug:    0,
+  info:     1,
+  warning:  2,
+  error:    3,
+  critical: 4,
+}
+
+class Logger
+  constructor: (source, logLvl=null, @formatter=kv.format, output=console.error) ->
+    if !logLvl?
+      strLogLvl = process.env.LOG_LEVEL_CONFIG
+      if !strLogLvl?
+        logLvl = LOG_LEVELS.Debug
+      else
+        vals = (val for key, val of LOG_LEVELS)
+        logLvl = if (strLogLvl in vals) then strLogLvl else LOG_LEVELS.Debug
+    @logLvl = logLvl
+    @globals = {}
+    @globals["source"] = source
+    @logWriter = output
+
+  setConfig: (source, logLvl, formatter, output) ->
+    @source = source
+    @logLvl = logLvl
```

Comment:
should we not verify the logLvl here just as we do in the constructor. Maybe one way to do this is to move out the validation logic into an internal method. 

then we could have in the `constructor`:

```coffeescript
strLogLvl = process.env.LOG_LEVEL_COFIG or LOG_LEVELS.Debug
logLvl = _validateLogLvl(strLogLvl)
// log the logLvl we are using
```

and in `setConfig`
```
@logLvl = _validateLogLvl(logLvl)
```